autobotAI_integrations/integrations/openai/ai_evaluator_code.py

```
@@ -1,15 +1,62 @@
+    try:
+        current_token_count = SYSTEM_PROMPT_TOKENS + estimate_tokens(prompt)
+        parsable_resources_count = 0
+        
+        for resource in resources:
+            resource_tokens = estimate_tokens(str(resource))
+            if current_token_count + resource_tokens < SAFE_INPUT_TOKENS:
+                parsable_resources_count += 1
+                current_token_count += resource_tokens
+            else:
+                logger.warning(f"Skipping resource due to token limit. Current: {current_token_count}, "
+                             f"Adding: {resource_tokens}, Safe limit: {SAFE_INPUT_TOKENS}")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)